### PR TITLE
[#173] Use refillable addresses

### LINKED ISSUE #173: Use auto-refillable addresses in network tests
# Clarification and motivation
Networks tests tend to fail due to insufficient balance errors. `cleveland` recently introduced auto-refillable addresses,
which are probably going to help us to avoid hardcoding transfer amounts in the code.

# Acceptance criteria
Network tests don't use hardcoded transfer amounts.

### PR DESCRIPTION
**Pull request by lierdakil**
## Description

Problem: We could stand to run smoke tests a little faster.

Solution: Batch some originations and transfers in smoke tests running
through Cleveland; modify tests running through tzbtc-client to
"emulate" batched transfers.

Problem: tests tend to error out with insufficient balance errors. We
hard-code some amounts in tests, but this is error-prone and not always
sufficient.

Solution: Use refillable addresses everywhere. This is basically free
for cases that don't need refills, and will mostlysave us from running
out of funds mid-test.

One particular hard-coded transfer has to be left in place because
that's running through tzbtc-client, and tzbtc-client doesn't do
refills.

**Discussion point**: The alternative to hard-coding a transfer, we could just as well set the "admin" to moneybag address in smoke tests. This would make the tests very slightly less robust, but it'll work without hard-coded transfer amounts.

## Related issue(s)

Resolves #173

#### Related changes (conditional)

- Tests (see [short guidelines](../blob/master/CONTRIBUTING.md#tests))
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - I checked whether I should update the docs and did so if necessary:
    - [ ] [README](../blob/master/README.md)
    - [ ] Haddock
    - [ ] [docs/](../blob/master/docs/)
  - [ ] I updated [changelog](../blob/master/ChangeLog.md) unless I am sure my changes are
        really minor.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../blob/master/docs/code-style.md).

### CODE REVIEW

**Comment by lierdakil** [comment]
Huh. Well, that apparently didn't work. Turns out some of our admin accounts are initialized in morley-ledgers-test. Unfortunate, but not impossible to work around.

File: test/Test/TZBTC.hs [hunk 1]
```diff
@@ -506,9 +506,10 @@ test_approvableLedgerV1 = testGroup "TZBTC contract approvable ledger tests"
       :: (MonadCleveland caps m)
       => ImplicitAddress -> AL.AlSettings -> m (ContractHandle (Parameter TZBTCv1) (Storage TZBTCv1) ())
     alOriginate = originateManagedLedger $ \op  -> do
-      let owner = case ML.opAdmin op of
+      let owner' = case ML.opAdmin op of
             MkAddress x@ImplicitAddress{} -> x
             _ -> error "Unexpected address type in alOriginate"
+      owner <- refillable $ pure owner' -- force owner to be refillable
```
Comment (pasqu4le) [inline test/Test/TZBTC.hs:512]: Why not
```suggestion
      owner <- case ML.opAdmin op of
            MkAddress x@ImplicitAddress{} -> refillable $ pure x -- force owner to be refillable
            _ -> error "Unexpected address type in alOriginate"
```

File: test/Test/TZBTC.hs [hunk 2]
```diff
@@ -527,9 +528,10 @@ test_approvableLedgerV2 :: TestTree
 test_approvableLedgerV2 = testGroup "TZBTC contract approvable ledger tests"
   [ AL.approvableLedgerGenericTest @(Parameter SomeTZBTCVersion) @(Storage SomeTZBTCVersion) $
     originateManagedLedger $ \op -> do
-      let owner = case ML.opAdmin op of
+      let owner' = case ML.opAdmin op of
             MkAddress x@ImplicitAddress{} -> x
             _ -> error "Unexpected address type in originateManagedLedger"
+      owner <- refillable $ pure owner' -- force owner to be refillable
```
Comment (pasqu4le) [inline test/Test/TZBTC.hs:534]: Same here

**Review by pasqu4le — APPROVED** [review]

File: test/Test/TZBTC.hs (hunk 1 above)
Comment (lierdakil) [reply test/Test/TZBTC.hs]: I've opted to move `refillable . pure` outside `case` to make `refillable` more obvious.

**Comment by lierdakil** [comment]
Eschewing changelog update as this only touches tests.

**Pull request merged**